### DISCUSSION TITLE
I have no name!

### DISCUSSION THREAD

**Discussion by jmcclelland** [ContainerSSH/ContainerSSH#541] [Questions] [answered]
I'm using containerSSH to allow users to ssh into a shared server without giving them direct access to the host itself. Each user has a valid user account and UID on the host. ContainerSSH authenticates them and then chooses which host directories to mount inside the container and launches the container with their UID . It's working well, except their username is not listed in `/etc/passwd` so their prompt says "I have no name!" and when they list directories, it shows UIDs instead of usernames.

I can't seem to run `adduser` when the container starts because they aren't root. 

Any suggestions on how to get their username into `/etc/passwd`?

**Comment by ghost** [comment, marked as answer]
Hey @jmcclelland,

There are several solutions:

1. Hard-code a user at a fixed UID and configure the container to run on that UID.
2. In Kubernetes, create an init container that seeds the username and share the `/etc/passwd` using a volume on an `emptyDir` or similar.
3. Add an init script, that when finished, starts the ContainerSSH agent and use that for your container to start (configure the `agentPath` accordingly.)

**Comment by jmcclelland** [comment]
Thank you so much! This is really helpful - i think the init script that starts the ContainerSSH agent is the way to go. I assume that it is run as root and that the ContainerSSH agent drops privileges?

**Reply by ghost** [reply]
No, the ContainerSSH agent doesn't drop privileges, you will have to implement that in your script.
